gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/avro/MRCompactorAvroKeyDedupJobRunner.java

```diff
@@ -99,7 +108,9 @@ protected void configureJob(Job job) throws IOException {
 
   private void configureSchema(Job job) throws IOException {
     Schema newestSchema = getNewestSchemaFromSource(job);
-    AvroJob.setInputKeySchema(job, newestSchema);
+    if (useSingleInputSchema) {
+      AvroJob.setInputKeySchema(job, newestSchema);
+    }
     AvroJob.setMapOutputKeySchema(job, this.shouldDeduplicate ? getKeySchema(job, newestSchema) : newestSchema);
```

Comment:
hmm, as far as I see, you can only easily set the outputkey schema in the Mapper, but for deduplication, this can differ from the inputschema due to `AvroJob.setMapOutputKeySchema(job, this.shouldDeduplicate ? getKeySchema(job, newestSchema) : newestSchema);` in `MRCompactorAvroKeyDedupJobRunner`.